cfme/tests/storage/test_volume_snapshot.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+
+from cfme import test_requirements
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.utils.log import logger
+from cfme.utils.wait import wait_for
+
+
+pytestmark = [
+    test_requirements.storage,
+    pytest.mark.ignore_stream("upstream"),
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.provider([OpenStackProvider], scope='module')
+]
+
+STORAGE_SIZE = 1
+
+
+@pytest.yield_fixture(scope='module')
+def snapshot(appliance, provider):
+    volume_collection = appliance.collections.volumes
+    snapshot_collection = appliance.collections.volume_snapshots.filter({'provider': provider})
+
+    manager_name = '{} Cinder Manager'.format(provider.name) if appliance.version >= '5.8' else None
+
+    # create new volume
+    volume = volume_collection.create(name=fauxfactory.gen_alpha(),
+                                      storage_manager=manager_name,
+                                      tenant=provider.data['provisioning']['cloud_tenant'],
+                                      size=STORAGE_SIZE,
+                                      provider=provider)
+
+    # create new snapshot for crated volume
+    snapshot_name = fauxfactory.gen_alpha()
+    volume.create_snapshot(snapshot_name)
+    snapshot = snapshot_collection.instantiate(snapshot_name, provider)
+
+    yield snapshot
+
+    try:
+        if snapshot.exists:
+            snapshot_collection.delete(snapshot)
+    except Exception as e:
+        logger.warning("{name}:{msg}: Snapshot deletion - skipping...".format(
+            name=type(e).__name__,
+            msg=str(e)))
+
+    try:
+        if volume.exists:
+            volume.delete(wait=False)
+    except Exception as e:
+        logger.warning("{name}:{msg} Volume deletion - skipping...".format(
+            name=type(e).__name__,
+            msg=str(e)))
+
+
+@pytest.mark.tier(3)
+def test_storage_volume_snapshot_create(snapshot):
+    wait_for(lambda: snapshot.status == 'available',
```

Comment:
**REQUIRED**: Where does this test create the snapshot? Whatever specific is tested should be in the test and not in the fixture, so you should put the `.create` in the test.